internal/cache/file/downloader/parallel_downloads_job.go

```diff
@@ -58,61 +60,158 @@ func (job *Job) downloadRange(ctx context.Context, dstWriter io.Writer, start, e
 		}
 	}()
 
-	monitor.CaptureGCSReadMetrics(ctx, util.Parallel, end-start)
+	monitor.CaptureGCSReadMetrics(ctx, util.Parallel, int64(end-start))
 
-	_, err = io.CopyN(dstWriter, newReader, end-start)
+	_, err = io.CopyN(dstWriter, newReader, int64(end-start))
 	if err != nil {
 		err = fmt.Errorf("downloadRange: error at the time of copying content to cache file %w", err)
 	}
 	return err
 }
 
+// RangeMap maintains the ranges downloaded by the different goroutines. This
```

Comment:
nit: Add a line - "If the ranges on merging completes a range starting 0, then update the offset in job's status"